src/components/Nipple.js

```diff
@@ -1,72 +1,63 @@
-import React, { useState } from 'react';
+import React, { useContext } from 'react';
 import ReactNipple from 'react-nipple';
-import DebugView from "react-nipple/lib/DebugView";
 import { makeStyles } from '@material-ui/core/styles';
+import { Context as ControllerContext } from '../context/Controller';
 
 const useStyles = makeStyles({
-    root: {
-        width: '100%',
-        height: '100%',
-    },
     nipple: {
-        width: '100%',
-        height: '100%',
+        width: '150px',
+        height: '150px',
         position: 'relative',
-        top: '50%',
-        left: '50%',
-        transform: 'translate(-50%, -50%)'
-    },
-    debug: {
-        position: 'absolute',
-        top: '0',
-        left: '0',
-        color: '#fff'
     }
 });
 
-function getPosition(radian) {
-    if (
-        radian > 0 && radian < 0.5 ||
-        radian > 5.5 && radian < 6.5
-    ) {
-        // return 'Right';
-        return { touchX: 1 }
-    }
-
-    if (radian > 0.5 && radian < 2.5) {
-        // return 'Up';
-        return { touchY: 1 }
-    }
-
-    if (radian > 2.5 && radian < 3.7) {
-        // return 'Left';
-        return { touchX: -1 }
-    }
-
-    if (radian > 3.7 && radian < 5.5) {
-        // return 'Down';
-        return { touchY: -1 }
-    }
-}
-
 export default function Nipple() {
     const classes = useStyles();
-    const [data, setData] = useState(undefined);
+    const { onControllerChange } = useContext(ControllerContext);
 
+    function setValue(values) {
+        console.log(values);
+        onControllerChange(values);
+    }
     const onMove = (evt, data) => {
-        const position = getPosition(data.angle.radian);
+        const force = Math.min(data.force, 1);
+        const degree = data.angle.degree;
+        if(degree <= 45) {
```

Comment:
![](https://media.giphy.com/media/l0IylOPCNkiqOgMyA/giphy.gif)

Don't ask but it works